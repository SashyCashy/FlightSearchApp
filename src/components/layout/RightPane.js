/**
 * File Description: This component is the right pane of the website to display the results of search
 * Author(s): Sashank Pindiproli
 * Date of Creation: 02/02/2020
 */

import React from 'react';
import styled from 'styled-components';
import FlightStatus from '../common/Status';
import FlightStatusDisplay from './FlightStatus/FlightStatusDisplay';

const RightPane = ({ to, from, fromDate, flightsList = [] }) => {
  let { direct, multiple } = flightsList;
  let count =
    direct !== undefined && multiple !== undefined
      ? direct.length + multiple.length
      : 0;
  return (
    <StyledContainer id="right-pane__container">
      <FlightStatus
        flightsCount={count}
        to={to}
        from={from}
        travelDate={fromDate}
      />

      {count > 0 ? <FlightStatusDisplay flightsList={flightsList} /> : null}
    </StyledContainer>
  );
};

const StyledContainer = styled.div`
  border: 2px solid black;
  width: calc(84.2%);
  margin-left: 1vh;
  margin-top: 0.5vh;
  overflow: scroll;
`;

export default RightPane;
