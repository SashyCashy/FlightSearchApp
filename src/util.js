const fetchCustomerList = async () => {
  let data = await fetch('/api/customers');
  let result = await data.json();
  return result;
};

const fetchCustomer = async (id) => {
  let data = await fetch(`/api/user/${id}`);
  let result = await data.json();
  return result;
};

export { fetchCustomerList, fetchCustomer };
