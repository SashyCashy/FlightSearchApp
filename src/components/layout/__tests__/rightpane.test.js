import React from 'react';
import { mount } from 'enzyme';
import RightPane from '../RightPane';
import { fetchCustomer } from '../../../util';

beforeEach(() => {
  fetch.resetMocks();
});

describe('Fetch data for right pane ', () => {
  it('Check the fetch call ', async () => {
    fetch.mockResponseOnce(
      JSON.stringify({
        name: 'James Cameroon',
        id: 7,
        age: '24',
        address: ['Maya Linda Rd San Diego CA 92126'],
        mage: 'male.png',
      })
    );

    let customer = await fetchCustomer(7);
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toHaveBeenCalledWith(`/api/user/7`);
    expect(customer).not.toBeNull();
    expect([customer]).toHaveLength(1);
  });

  test('Check if the values are being fetched or not ', async () => {
    fetch.mockResponseOnce(
      JSON.stringify({
        name: 'James Cameroon',
        id: 7,
        age: '24',
        address: ['Maya Linda Rd San Diego CA 92126'],
        image: 'male.png',
      })
    );

    let customerDetails = await fetchCustomer(7);

    const wrapper = mount(<RightPane customerDetails={customerDetails} />);
    expect(wrapper).toMatchSnapshot();
  });
});
