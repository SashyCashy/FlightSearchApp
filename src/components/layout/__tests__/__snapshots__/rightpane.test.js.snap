// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch data for right pane  Check if the values are being fetched or not  1`] = `
<div
  className="sc-AxjAm eWrtAF"
  id="right-pane__container"
>
  <div
    className="sc-AxirZ QTrgE card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        James Cameroon
      </div>
      <img
        className="card-img"
        src="male.png"
      />
      <p
        className="card-text"
      >
        Maya Linda Rd San Diego CA 92126
      </p>
    </div>
  </div>
</div>
`;
